src/tasks/trainerUtils.py

```diff
@@ -530,46 +557,74 @@ def parse_trainer_btl_set(substring):
-    regex_lookup = regex_lookup_dict.get(type)
+def parse_ev_script_file(file_path):
+    """
+    The purpose of this function is to parse a text file and find every instance of the substring _TRAINER_BTL_SET or _TRAINER_MULTI_BTL_SET.
+    """
+
+    def get_zoneID(areaName):
+        for places in areas:
+            if areaName in places:
+                zoneID = int(areas.index(places) - 1)
+                return zoneID
+
+    trainers = []
 
     with open(file_path, 'r', encoding="utf-8") as f:
-        match_count = 0
-        found_lookup = False
-        trainers = []
         for line in f:
-            if match_count == count:
-                break
             substrings = line.split('\n')
             for substring in substrings:
-                if match_count == count:
-                    break
-                if not found_lookup:
-                    if substring.startswith(lookup) and REMATCH_SUBSTRING not in substring:
-                        found_lookup = True
+                areaName = file_path.split("/")[-1].split(".")[0].upper()
+                zoneID = get_zoneID(areaName)
+                
+                if TRAINER_BATTLE in substring or MULTI_TRAINER_BATTLE in substring:
+                    args = parse_trainer_btl_set(substring.strip())
                 else:
-                    if regex_lookup in substring:
-                        match = re.split(ldval_pattern, substring)[2]
-                        if match:
-                            trainer_id = match.strip("'")
-                            trainers.append(trainer_id)
-                            match_count += 1
+                    continue
+
+                if zoneID is None or zoneID < 0:
+                    print("The trainers in this area are currently not supported:", zoneID, areaName, args[0])
+                    raise UnsupportedTrainer
+
+                if -1 in args:
+                    print("There is something wrong with the args from this area", zoneID, areaName, args[0])
+                    raise InvalidArg
+
+                trainers.extend(get_all_trainer_data(file_path, areaName, zoneID, args))
+                
     return trainers
 
+def process_files(folder_path, callback):
+    """
+    Calls the provided callback function on every file in the specified folder.
+    
+    :param folder_path: The path to the folder containing the files to be processed.
+    :param callback: A function that takes a file path as its only argument and performs some action on it.
+    """
+    trainers_list = []
+    trainers = []
+    trainers_set = set()
+    for filename in os.listdir(folder_path):
+        file_path = os.path.join(folder_path, filename)
+        try:
+            trainers = callback(file_path)
+        except (FileNotFoundError, IsADirectoryError):
+            print(f"{file_path} is not a valid file path or does not exist")
+            return
+        except (UnsupportedTrainer, MissingData):
+            continue
```

Comment:
Do you really want to continue in the event of unhandled trainer data?

Reply:
Yes, because this more like a warning rather than an outright error. This is mostly to find out where there is still data but it's not in a specific area. I should probably put that data somewhere tho shouldn't I?

Reply:
Although, I should probably error it out with the MissingData. Changing it now.